### PR TITLE
[TS] Migrate Avatar tests to .tsx

### PR DESCRIPTION
**Pull request by alexmirrington**
#### What does this PR do?

This PR migrates Avatar component tests from `*.js` to `*.tsx` to allow `ts-jest` to catch type-related errors in the unit testing pipeline. A few changes to the tests had to be made to support this, and are justified below.

#### Where should the reviewer start?

- Ensure all tests are currently passing
- Refer to the background context section below for more information about the initial errors picked up by TypeScript and the fixed implemented in this PR.

#### What testing has been done on this PR?

- Manually ran and passed Anchor component tests locally.
- CircleCI tests have passed.

#### How should this be manually tested?

- The reviewer should verify all changes made to tests and ensure that:
    - All test removals are due to unused legacy code and will not cause regressions
    - All tests behave identically and modifications are type-related.

#### Any background context you want to provide?

Upon initial renaming of tests, the following errors were picked up by `ts-jest`:

Input: `$ yarn test Avatar`
Output: 
```
yarn run v1.22.10

$ jest --runInBand Avatar
 FAIL  src/js/components/Avatar/__tests__/Avatar-test.tsx
  ● Test suite failed to run

    src/js/components/Avatar/__tests__/Avatar-test.tsx:19:30 - error TS2322: Type '{ id: string; name: string; }' is not assignable to type 'IntrinsicAttributes & AvatarExtendedProps & { children?: ReactNode; }'.
      Property 'name' does not exist on type 'IntrinsicAttributes & AvatarExtendedProps & { children?: ReactNode; }'.

    19         <Avatar id="test id" name="test name" />
                                    ~~~~
    src/js/components/Avatar/__tests__/Avatar-test.tsx:90:17 - error TS2322: Type 'Element' is not assignable to type 'string'.

    90         <Avatar src={<Favorite color="accent-2" />} background="accent-4" />
                       ~~~

      src/js/components/Avatar/index.d.ts:6:3
        6   src?: string;
            ~~~
        The expected type comes from property 'src' which is declared here on type 'IntrinsicAttributes & AvatarExtendedProps & { children?: ReactNode; }'

Jest: "global" coverage threshold for statements (86.9%) not met: 0%
Jest: "global" coverage threshold for branches (76.3%) not met: 0%
Jest: "global" coverage threshold for lines (87.9%) not met: 0%
Jest: "global" coverage threshold for functions (88%) not met: 0%
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        12.16 s
Ran all test suites matching /Avatar/i.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

- To resolve the error on line 19, i updated the test spec to remove the name attribute from the underlying `div`, since `div` tags [shouldn't contain name attributes](https://www.w3schools.com/tags/att_name.asp).
- The error on line 90 occurs because `src` should contain a string representing an image URL (_e.g._ from gravatar). As [per the official Grommet docs](https://v2.grommet.io/avatar), if we want to use an icon we should use `props.children`. I've updated the test and spec to capture this behaviour.

#### What are the relevant issues?

Parent tracking issue: #5599

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

No, all fixes are test-related and not component-related.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

**Review by jcfilben — APPROVED** [review]

File: src/js/components/Avatar/__tests__/Avatar-test.tsx [hunk 1]
```diff
@@ -87,7 +87,9 @@ describe('Avatar', () => {
   test('icon renders', () => {
     const { container } = render(
       <Grommet>
-        <Avatar src={<Favorite color="accent-2" />} background="accent-4" />
```
Comment (ericsoderberghp) [inline src/js/components/Avatar/__tests__/Avatar-test.tsx:90]: We need both, a test with an element `src` as well as children. Taking a look at propTypes.js, it looks like index.d.ts is missing a type for `src`.

Comment (jcfilben) [reply src/js/components/Avatar/__tests__/Avatar-test.tsx:90]: In the documentation on grommet-site `src`'s only type is a string and this is reflected in the index.d.ts. However, the propTypes file has `src: PropTypes.oneOfType([PropTypes.string, PropTypes.element])` so I think it is the propTypes file that is incorrect.

I tested `<Avatar src={<Favorite color="accent-2" />} background="accent-4" />` and it wasn't showing up correctly. Whereas the code below was showing up correctly. [Example here](https://codesandbox.io/s/grommet-v2-template-forked-z91x2?file=/index.js).
```
<Avatar background="accent-4">
  <Favorite color="accent-2" />
</Avatar>
```

Comment (ericsoderberghp) [reply src/js/components/Avatar/__tests__/Avatar-test.tsx:90]: OK. I had just noticed the discrepancy. Thanks for delving deeper.

Comment (ericsoderberghp) [reply src/js/components/Avatar/__tests__/Avatar-test.tsx:90]: Still need to clean up `propTypes.js`

Comment (alexmirrington) [reply src/js/components/Avatar/__tests__/Avatar-test.tsx:90]: To add to this, the old snapshot never contained an SVG for the icon, where now it does. I'll remove `PropTypes.element` from the `propTypes` file.

**Review by jcfilben — APPROVED** [review]

**Pull request merged**